jax/_src/interpreters/pxla.py

```diff
@@ -2339,8 +2341,8 @@ def lower_sharding_computation(
    nreps, tuple_args, shape_poly_state) = _cached_lowering_to_hlo(
        closed_jaxpr, api_name, fun_name, backend, semantic_in_shardings,
        semantic_out_shardings, in_layouts, out_layouts, num_devices,
-       tuple(da_object) if prim_requires_devices else None, donated_invars,
-       name_stack, all_default_mem_kind, inout_aliases,
+       tuple(da_object) if prim_requires_devices is not None else None,  # type: ignore[arg-type]
```

Comment:
prim_require_devices is a bool. It will never be None.

Can you just keep the logic as is please and ignore the type checker? The current change is incorrect.

Reply:
https://github.com/jax-ml/jax/blob/main/jax/_src/dispatch.py#L216 -- this returns a bool